src/js/charts/line.js

```diff
@@ -938,4 +938,336 @@
   }
 
   MG.register('line', lineChart);
+    
+    if (args.brushing) {
+      
```

Comment:
@wlach basically, these args are showing as undefined and THIS IS THE MAIN PROBLEM. What can be done now in that case ?

Reply:
So this wasn't quite what I meant. I meant you would literally put the blocks of code that are called via hooks in the appropriate place in the line chart method. Take, for example, the `after_init` method:

https://github.com/dandehavilland/mg-line-brushing/blob/e762a56d516f1dc9ce4bcd0180f18703d456bc4d/src/js/hooks.js#L269

You would look up where we are calling the hook:

https://github.com/mozilla/metrics-graphics/blob/20a661f13457a971fbbb757f8fa3c42f0d4a0d0a/src/js/charts/line.js#L818

And then call/apply the code that would have been called by the hook directly (only, of course, if the brushing argument is enabled):

https://github.com/dandehavilland/mg-line-brushing/blob/e762a56d516f1dc9ce4bcd0180f18703d456bc4d/src/js/hooks.js#L66

In this case, I would probably turn the brushing function into a `this` method to avoid the need to call apply and pass the chartContext:

https://github.com/mozilla/metrics-graphics/blob/20a661f13457a971fbbb757f8fa3c42f0d4a0d0a/src/js/charts/line.js#L823

So, yes, in one sense this is mostly a matter of just moving code around -- but you will need to modify it some as well. As you can see in the line chart methods, we already have all sorts of code to add various features to the chart and make things behave appropriately as the user is interacting with it -- this is just about adding one more (with the bonus of having an existing code base to take from). So if you have a doubt, just look at the rest of the code in this file for inspiration.